### PR TITLE
[SDK-2435] Add Organizations support to Management API Client

### PR DESCRIPTION
**Pull request by evansims**
This pull request adds support for Organizations to the Management API Client:

- Adds `Management::organizations()` singleton provider.
- Adds `Users::getOrganizations()` method for retrieving organizations user is a member of.
- Updates `Auth0\SDK\API\Management\Tickets` class:
  - `Tickets::createEmailVerificationTicket()` now supports passing `organization_id` param. Fixed `client_id` not getting passed properly.
  - `Tickets::createPasswordChangeTicket()` now has an optional `organization_id` argument.
  - `Tickets::createPasswordChangeTicketByEmail()` now has an optional `organization_id` argument.
  - `Tickets::createPasswordChangeTicketRaw()` now has an optional `organization_id` argument.
- Updates `Auth0\SDK\API\Management\Jobs` class:
  - `Jobs::sendVerificationEmail()` now supports passing `organization_id` param.
- Adds `Auth0\SDK\API\Management\Organizations` class:
  - `Organizations::create()` creates a new organization.
  - `Organizations::update()` makes changes to an existing organization.
  - `Organizations::delete()` deletes an organization.
  - `Organizations::getAll()` returns an array of all created organizations. Pagination supported.
  - `Organizations::get()` returns details about an existing organization, by its `id`.
  - `Organizations::getByName()` returns details about an existing organization, by its `name`.
  - `Organizations::getEnabledConnections()` returns an array of all enabled connections. Pagination supported.
  - `Organizations::getEnabledConnection()` returns details about an enabled connection.
  - `Organizations::addEnabledConnection()` attachs a connection to an organization.
  - `Organizations::updateEnabledConnection()` makes changes to properties of a connection on an organization.
  - `Organizations::removeEnabledConnection()` detaches a connection from an organization.
  - `Organizations::getMembers()` returns an array of all users who are members of an organization. Pagination supported.
  - `Organizations::addMembers()` makes one or more users members of an organization.
  - `Organizations::removeMembers()` removes membership of one or more users from an organization.
  - `Organizations::getMemberRoles()` returns an array of roles a user has been granted within the organization.
  - `Organizations::addMemberRoles()` grants a user one or more roles within an organization.
  - `Organizations::removeMemberRoles()` strips one or more roles from a member of an organization.
  - `Organizations::getInvitations()` returns an array of all pending invitations for an organization. Pagination supported.
  - `Organizations::getInvitation()` returns details about a pending invitation for an organization.
  - `Organizations::createInvitation()` creates an invitation to join an organization.
  - `Organizations::deleteInvitation()` deletes a pending invitation for an organization.
- Adds `GenericResource::checkEmptyOrInvalidArray()` helper.
- Updates `UsersMockedTest` for new `Users::getOrganizations()` method.
- Adds `OrganizationsTest` unit tests for new `Auth0\SDK\API\Management\Organizations` class.
- Adds `OrganizationsIntegrationTest` integration tests for new `Auth0\SDK\API\Management\Organizations` class.

### CODE REVIEW

**Comment by evansims** [comment]
> I can see the integration tests only run on the master branch, were you able to run them manually as part of this work and if so presumably they were ok?

@stevehobbsdev Yup! Just pushed a few small tweaks, but it's all green there

```
➜  auth0-php git:(feat-sdk-2435) ✗ php vendor/bin/phpunit --testsuite integration --group management
PHPUnit 9.5.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.27
Configuration: /Users/evan/Projects/auth0-PHP/phpunit.xml.dist

...........                                                       11 / 11 (100%)

Time: 00:47.012, Memory: 8.00 MB

OK (11 tests, 33 assertions)
```

**Comment by evansims** [comment]
@stevehobbsdev Sorry for the 11th-hour change, but just noticed I had overlooked passing the new `organization_id` param on the job/ticket endpoints, and we were missing `client_id` on one. Fixed that.

**Comment by evansims** [comment]
Added integrations tests for invitations as well, since I figured out what was blocking me on those. That's everything I had on my todo list here.

File: src/API/Management/Organizations.php [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+            ->withDictParams($this->normalizeRequest($params))
+            ->call();
+    }
+
+    /**
+     * Get a connection (by ID) associated with an organization.
+     * Required scope: "read:organization_connections"
+     *
+     * @param string              $organization Organization (by ID) to list connections of.
+     * @param array<string,mixed> $params       Optional. Additional options to include with the request, such as pagination or filtering parameters.
```
Comment (stevehobbsdev) [inline src/API/Management/Organizations.php:204]: Should this instead refer to `connection` rather than `params`? Looks like this is for another method.

**Review by stevehobbsdev — CHANGES_REQUESTED** [review]
All looks good, but noticed that one of the doc comments looks to be incorrect.

File: src/API/Management/Organizations.php (hunk 1 above)
Comment (evansims) [reply src/API/Management/Organizations.php]: Whoops, good catch! Fixed 👍

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**